mobile/cmp/navigator/NavigatorModel.js

```diff
@@ -4,78 +4,161 @@
  *
  * Copyright © 2018 Extremely Heavy Industries Inc.
  */
-import {HoistModel} from '@xh/hoist/core';
-import {observable, action} from '@xh/hoist/mobx';
-import {clone} from 'lodash';
+import {XH, HoistModel, elem} from '@xh/hoist/core';
+import {bindable, observable, action} from '@xh/hoist/mobx';
+import {throwIf} from '@xh/hoist/utils/js';
+import {uniqBy, keys, find, merge, isEqual} from 'lodash';
 
 import {NavigatorPageModel} from './NavigatorPageModel';
 
 /**
- * Model for handling navigation between Onsen pages
+ * Model for handling stack-based navigation between Onsen pages.
+ * Provides support for routing based navigation.
  */
 @HoistModel
 export class NavigatorModel {
-    initPageModel = null;
-    @observable title;
+    /** @member {string} */
+    @bindable title;
+
+    /** @member {NavigatorPageModel[]} */
     @observable.ref pages = [];
 
+    /** @member {Object[]} */
+    routes = [];
+
     _navigator = null;
     _callback = null;
 
     /**
-     * @param {Object} page - configuration for NavigatorPageModel
+     * {Object[]} routes - configs for NavigatorPageModels.
      */
-    constructor(page) {
-        this.initPageModel = new NavigatorPageModel(page);
-        this.onPageChange();
-    }
+    constructor({routes}) {
+        throwIf(routes.length === 0, 'NavigatorModel needs at least one route.');
+        throwIf(routes.length !== uniqBy(routes, 'id').length, 'One or more routes in NavigatorModel has a non-unique id.');
+        this.routes = routes;
 
-    /**
-     * @param {Object} page - configuration for NavigatorPageModel
-     * @param {function} [callback] - function to execute after the page transition
-     */
-    pushPage(page, callback) {
-        this._navigator.pushPage(new NavigatorPageModel(page));
-        this._callback = callback;
+        this.addReaction({
+            track: () => XH.routerState,
+            run: () => this.onRouteChange()
```

Comment:
minor point, but these run functions can all just be direct references to the method, and don fat arrow functions.  `addReaction()` already calls the run function as a method on `this` by default...